src/DeliveryCartReferencedContent.php

```diff
@@ -0,0 +1,155 @@
+<?php
+
+
+namespace Drupal\delivery;
+
+use Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException;
+use Drupal\Component\Plugin\Exception\PluginNotFoundException;
+use Drupal\delivery\Entity\MenuLinkContent;
+use Drupal\Core\Entity\EntityInterface;
+
+class DeliveryCartReferencedContent {
+
+  static $count = 0;
+
+
+  /**
+   * Looks for any referenced menu items and adds them to the cart.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *
+   * @return false|int
+   */
+  public static function addMenuItems(EntityInterface $entity){
+    if(!self::isNode($entity)) {
+      return FALSE;
+    }
+
+    $node_id = $entity->id();
+    $menu_link_manager = \Drupal::service('plugin.manager.menu.link');
+    $routes = $menu_link_manager->loadLinksByRoute('entity.node.canonical', ['node' => $node_id]);
+
+    if(!empty($routes)) {
+      foreach ($routes as $menuLink){
+        $plugin = $menuLink->getPluginDefinition();
+        if(isset($plugin) && isset($plugin['metadata']) && isset($plugin['metadata']['entity_id'])){
+          $id = $plugin['metadata']['entity_id'];
+          $linkEntity = MenuLinkContent::load($id);
+          if($linkEntity){
+            \Drupal::service('delivery.cart')->addToCart($linkEntity);
+            self::$count++;
+          }
+        }
+      }
+    }
+
+    return self::$count;
+  }
+
+  /**
+   * Looks for any referenced media in the body and adds them to the cart.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *
+   * @return false|int
+   */
+  public static function addMediaItems(EntityInterface $entity) {
+    if(!self::isNode($entity)) {
+      return FALSE;
+    }
+
+    if(!self::hasField($entity,'body')) {
```

Comment:
Hi @elistone !
This code would actually belong to the ckeditor5_sections module. And it should not search only in the body field, but basically in any field which is of Sections type.

Reply:
Moved over to ckeditor_sections see: AmazeeLabs/ckeditor5_sections#124